cql3/selection/selection.cc

```diff
@@ -670,7 +670,8 @@ bool result_set_builder::restrictions_filter::do_filter(const selection& selecti
             // fallthrough
         case column_kind::regular_column: {
             if (cdef->kind == column_kind::regular_column && !row_iterator) {
-                continue;
+                // Since we don't have IS NULL, no restriction on a regular column can be satisfied if the column is NULL (#20485)
+                return false;
```

Comment:
A thought: Is the optimization of not calling the expression (because it can't possibly return true) worth it? You could have also run the actual expression and get from it always a "false" as a response.

In any case, you are right that because we don't have IS NOT or other things that might make a true (constants, NOT, etc.) the "return false" here is correct. But it creates a risk that one day in the future it will stop being corect.

Reply:
In #20487 the limitation is gone.